client-index/src/service/address_service.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
+
+use parity_codec::{Decode, Encode};
+
+use chain_core::init::coin::Coin;
+use chain_core::tx::data::address::ExtendedAddr;
+use chain_core::tx::data::input::TxoPointer;
+use chain_core::tx::data::output::TxOut;
+use chain_core::tx::data::TxId;
+use client_common::balance::TransactionChange;
+use client_common::{ErrorKind, Result, Storage};
+
+const KEYSPACE: &str = "index_address";
+
+/// All the details related to an address
+#[derive(Debug, Encode, Decode)]
+pub struct AddressDetails {
+    /// Unpent transactions corresponding to an address
+    pub unspent_transactions: Vec<(TxoPointer, TxOut)>,
```

Comment:
I guess similarly to the note below, this would ideally change to BTreeMap?

Reply:
Added a comment to change it when `BTreeSet` is implemented in `parity_codec`.